### PR TITLE
Allow to configure public/temporary url generation

### LINKED ISSUE #118: Please add public_url option to filesystem configuration
https://flysystem.thephpleague.com/docs/usage/public-urls/#prefix-public-url-generation

### PR DESCRIPTION
**Pull request by norkunas**
I've added 3 new options:
- `public_url` which allows to use a single url or multiple urls and based `flysystem` will resolve public url generator;
- `public_url_generator` - a string value of service id;
- `temporary_url_generator` - a string value of service id.

and also added tests and fixed some deprecation in tests.

Closes #118
Replaces #119

### CODE REVIEW

**Comment by COil** [comment]
Hi, is there something preventing from merging this? /cc @tgalopin

File: tests/Kernel/config/framework.yaml [hunk 1]
```diff
@@ -1,3 +1,4 @@
 framework:
     secret: 'test_secret'
     test: true
+    http_method_override: false
```
Comment (tgalopin) [inline tests/Kernel/config/framework.yaml:4]: What's the reason for this change?

**Review by tgalopin — COMMENTED** [review]
No reason other than I missed it :) thanks for the ping

A small question but otherwise it seems good to me

File: tests/Kernel/config/framework.yaml (hunk 1 above)
Comment (norkunas) [reply tests/Kernel/config/framework.yaml:4]: To fix deprecation notice

**Review by tgalopin — APPROVED** [review]

**Comment by tgalopin** [comment]
Thanks @norkunas !

**Comment by COil** [comment]
Thanks for the work. If we can have a tag with this, that would be great. 🙂

**Comment by COil** [comment]
Can we have a new tag @tgalopin  please? I have to fork the bundle to use the new 'public_url' feature for now:  https://github.com/coopTilleuls/flysystem-bundle/tree/3.1.0

**Comment by tgalopin** [comment]
Here it is :) : https://github.com/thephpleague/flysystem-bundle/releases/tag/3.1.0

**Comment by COil** [comment]
Thanks @tgalopin ! :)
